### PR TITLE
Pause and Resume Functionality

### LINKED ISSUE #127: Specify details about the paue & resume functionality

### PR DESCRIPTION
**Pull request by jGleitz**
More details on the pause & resume functionality, including additions to the “Data” chapter.

Please check whether you think this closes #127 for you.

### CODE REVIEW

File: doc/Requirements Specification/Parts/Functional Requirements.lyx [hunk 1]
```diff
@@ -538,12 +594,12 @@ gls{component}
...
 \begin_inset CommandInset label
 LatexCommand label
-name "/OF120/"
+name "/OF180/"
 
 \end_inset
 
  If requested by the user, Beagle shuts down the computer it's running on
- after it finished.
+ after it finished a measurement.
```
Comment (flyfl) [inline doc/Requirements Specification/Parts/Functional Requirements.lyx]: This sentence doesn't makes sense. It should be something like "If reqeusted by the user, Beagle shuts down the computer when/after a measurment has been finished."

Comment (jGleitz) [inline doc/Requirements Specification/Parts/Functional Requirements.lyx]: “the computer” is very general and I tried to be precise. Is "If reqeusted by the user, Beagle shuts down the computer it’s running on after all measurements finished." okay?

**Comment by flyfl** [comment]
lgtm. Could you take a look at my remark?

**Comment by jGleitz** [comment]
> Could you take a look at my remark?

I did, see above.

File: doc/Requirements Specification/Parts/Functional Requirements.lyx (hunk 1 above)
Comment (flyfl) [inline doc/Requirements Specification/Parts/Functional Requirements.lyx]: "shuts down the computer it's running on" sounds strange. Is that proper usage of English?

**Comment by RomanLangrehr** [comment]
[PDF](http://jgleitz.github.io/Beagle/branches/pause-resume/Requirements%20Specification.pdf)

File: doc/Requirements Specification/Parts/Functional Requirements.lyx (hunk 1 above)
Comment (jGleitz) [inline doc/Requirements Specification/Parts/Functional Requirements.lyx]: I really think so.

Do you have a better alternative?

**Comment by RomanLangrehr** [comment]
What about joining /OF130/ and /OF140/? Because both of them make no sense without the other.

File: doc/Requirements Specification/Parts/Functional Requirements.lyx (hunk 1 above)
Comment (flyfl) [inline doc/Requirements Specification/Parts/Functional Requirements.lyx]: What do you want to say? The computer keeps still running during a measurement, while the shut down request is marked?

**Comment by jGleitz** [comment]
> What about joining /OF130/ and /OF140/? Because both of them make no sense without the other.

I see. How about making `/OF130/` “pause and resume” and `/OF140/`“pause and resume after shutdown”, as the latter might require significantly more work?

**Comment by acBerger** [comment]
> I see. How about making /OF130/ “pause and resume” and /OF140/“pause and resume after shutdown”, as the latter might require significantly more work?

I like this idea.

**Comment by RomanLangrehr** [comment]
> /OF140/ “pause and resume after shutdown”, as the latter might require significantly more work?

What do you mean? How can our softwre be paused when the PC is shut down?!

**Comment by jGleitz** [comment]
> What do you mean? How can our softwre be paused when the PC is shut down?!

You know I like to be unprecise ;) I mean “The user may resume paused measurements even if the computer was turned off between pausing and resuming”

**Comment by RomanLangrehr** [comment]
> You know I like to be unprecise ;) I mean “The user may resume paused measurements even if the computer was turned off between pausing and resuming”

And what is more difficult when the PC has been shut down between pause and resume?

**Comment by jGleitz** [comment]
> And what is more difficult when the PC has been shut down between pause and resume?

If we don’t require that, we could simply tell our worker process to sleep and leave everything in memory. That’s easy to implement (and might still be better than no pause feature at all).

If the user has shut down the PC, we need true persistence.

**Comment by RomanLangrehr** [comment]
> If the user has shut down the PC, we need true persistence.

We already need true persistence, when the user e.g. closes eclipse or something like that.
I agree, that this will be more difficult, but I won't specify it only for PC shutdown, but with "closing Beagle".

**Comment by jGleitz** [comment]
Okay, so we agree on `/OF140/` to be (roughly) “resume works after Beagle was closed”?

**Comment by RomanLangrehr** [comment]
> Okay, so we agree on /OF140/ to be (roughly) “resume works after Beagle was closed”?

Exactly.

**Comment by acBerger** [comment]
> Okay, so we agree on /OF140/ to be (roughly) “resume works after Beagle was closed”?

I do

File: doc/Requirements Specification/Parts/Functional Requirements.lyx (hunk 1 above)
Comment (jGleitz) [inline doc/Requirements Specification/Parts/Functional Requirements.lyx]: No, “the computer [Beagle] is running on” specifies which computer we mean. Not any computer, but the exact computer Beagle runs on.

**Comment by jGleitz** [comment]
I implemented the changes as discussed.

**Comment by acBerger** [comment]
lgtm

File: doc/Requirements Specification/Parts/Functional Requirements.lyx (hunk 1 above)
Comment (jGleitz) [inline doc/Requirements Specification/Parts/Functional Requirements.lyx]: @flyfl, what do we do with this sentence? What do you thing, @RomanLangrehr, @acBerger, @m1cm1c?

Comment (acBerger) [inline doc/Requirements Specification/Parts/Functional Requirements.lyx]: I understand the problem of @flyfl, but I don't think it is very bad. I don't have a better wording.
Maybe following is better: "After finishing the measurements, Beagle is able to shut down the computer it used, if requested by the user."
But I am afraid it is just the same problem in other words.

Comment (RomanLangrehr) [inline doc/Requirements Specification/Parts/Functional Requirements.lyx]: I think we can keep this sentence. Maby we just change `a measurment` to `the measurments`, as beagle can measure c component multiple times.

Comment (jGleitz) [inline doc/Requirements Specification/Parts/Functional Requirements.lyx]: @RomanLangrehr: 

> Is "If requested by the user, Beagle shuts down the computer it’s running on after all measurements finished." okay?

Comment (RomanLangrehr) [inline doc/Requirements Specification/Parts/Functional Requirements.lyx]: Definitly

**Comment by RomanLangrehr** [comment]
Lgtm

**Pull request merged**